python/toolbox/system_cpu_topology.py

```diff
@@ -0,0 +1,273 @@
+import copy
+from pathlib import Path
+
+def _debug_log(log_msg):
+    return(print("DEBUG: %s" % (log_msg)))
+
+class system_cpu:
+    def __init__(self, cpu_dir, debug = False):
+        self.debug = debug
+
+        if self.debug:
+            _debug_log("system_cpu: processing %s" % (cpu_dir))
+
+        partition = str(cpu_dir).partition('cpu/cpu')
+        if partition[2] == '':
+            raise AttributeError("system_cpu: Could not extract cpu_id from '%s'" % (cpu_dir))
+        self.cpu_id = int(partition[2])
+
+        if self.debug:
+            _debug_log("system_cpu: found cpu=%s" % (self.cpu_id))
+
+        file = cpu_dir / 'online'
+        if file.exists() and file.is_file():
+            with file.open() as fh:
+                self.online = int(fh.readline().rstrip())
+                if self.debug:
+                    _debug_log("system_cpu: found cpu=%s online=%s" % (self.cpu_id, self.online))
+        else:
+            raise AttributeError("system_cpu: Could not open 'online' file for cpu=%d" % (cpu_id))
+
+        dir = cpu_dir / 'topology'
+        if dir.exists() and dir.is_dir():
+            for file in dir.iterdir():
+                partition = str(file).partition('topology/')
+                if partition[2] != '':
+                    if partition[2] == 'physical_package_id':
```

Comment:
this huge if/else can be replaced with a dictionary based 'switch case' block.
def f1():
    print("f1")

def f2():
    print("f2")

function_select = {
    'core_id': f1,
    'die_id: f2
}

function_select[partition[2]]()